src/validation/variable.rs

```diff
@@ -31,6 +42,47 @@ pub fn visit_variable(validator: &mut Validator, variable: &Variable, context: &
     visit_data_type_declaration(validator, &variable.data_type_declaration, context);
 }
 
+/// Validates Variable Length Arrays as specified in the IEC61131-3, i.e. VLAs are only allowed to be defined
+/// inside the following Variable Block and POU combinations
+/// - Input, Output and InOut within a Function or Method or
+/// - InOut within Function-Block
+fn validate_vla(validator: &mut Validator, pou: Option<&Pou>, block: &VariableBlock, variable: &Variable) {
+    let Some(pou) = pou else {
+        if matches!(block.variable_block_type, VariableBlockType::Global) {
+            validator.push_diagnostic(Diagnostic::invalid_vla_container(
+                "VLAs can not be defined as global variables".to_string(),
+                variable.location.clone())
+            )
+        }
+
+        return;
+    };
+
+    match (&pou.pou_type, block.variable_block_type) {
+        (PouType::Function, VariableBlockType::Input(ArgumentProperty::ByVal)) => {
+            validator.push_diagnostic(Diagnostic::vla_by_val_warning(variable.location.clone()))
+        }
```

Comment:
does this warning match block now steal the chance to do the bottom validation? (line 71..82) (invalid vla_container)?

Reply:
The warning is specifically for the combination of VAR_INPUT (ByVal) in a FUNCTION, in which case no other validation messages are necessary/possible